evaluate/repository.go

```diff
@@ -19,34 +19,42 @@ import (
 	"github.com/symflower/eval-dev-quality/util"
 )
 
-// Repository evaluate a repository with the given model and language.
-func Repository(logger *log.Logger, resultPath string, model evalmodel.Model, language language.Language, testDataPath string, repositoryName string) (repositoryAssessment metrics.Assessments, problems []error, err error) {
-	log, logClose, err := log.WithFile(logger, filepath.Join(resultPath, evalmodel.CleanModelNameForFileSystem(model.ID()), language.ID(), repositoryName+".log"))
+// Repository holds data about a repository.
+type Repository struct {
+	// Name holds the name of the repository.
+	Name string
+	// DataPath holds the path to the repository.
+	DataPath string
+}
+
+// Evaluate evaluates a repository with the given model and language.
+func (r *Repository) Evaluate(logger *log.Logger, resultPath string, model evalmodel.Model, language language.Language) (repositoryAssessment metrics.Assessments, problems []error, err error) {
+	log, logClose, err := log.WithFile(logger, filepath.Join(resultPath, evalmodel.CleanModelNameForFileSystem(model.ID()), language.ID(), r.Name+".log"))
 	if err != nil {
 		return nil, nil, err
 	}
 	defer logClose()
 
-	log.Printf("Evaluating model %q using language %q and repository %q", model.ID(), language.ID(), repositoryName)
+	log.Printf("Evaluating model %q using language %q and repository %q", model.ID(), language.ID(), r.Name)
 	defer func() {
-		log.Printf("Evaluated model %q using language %q and repository %q: encountered %d problems: %+v", model.ID(), language.ID(), repositoryName, len(problems), problems)
+		log.Printf("Evaluated model %q using language %q and repository %q: encountered %d problems: %+v", model.ID(), language.ID(), r.Name, len(problems), problems)
 	}()
 
-	filePaths, err := language.Files(log, testDataPath)
+	filePaths, err := language.Files(log, r.DataPath)
 	if err != nil {
 		return nil, problems, pkgerrors.WithStack(err)
 	}
 
 	repositoryAssessment = metrics.NewAssessments()
 	for _, filePath := range filePaths {
-		if err := ResetTemporaryRepository(logger, testDataPath); err != nil {
+		if err := r.Reset(logger); err != nil {
 			logger.Panicf("ERROR: unable to reset temporary repository path: %s", err)
 		}
```

Comment:
Usability nitpick: since we anyways panic here (and I think we always panic if the reset mechanism fails) we could also just panic within `Reset`.

Reply:
Nah, not a friend of that. It should be the caller's decision what to do with the error.